lib/mongoid/association/accessors.rb

```diff
@@ -43,7 +43,17 @@ def __build__(name, object, association, selected_fields = nil)
       def create_relation(object, association, selected_fields = nil)
         type = @attributes[association.inverse_type]
         target = association.build(self, object, type, selected_fields)
-        target ? association.create_relation(self, target) : nil
+        target = target ? association.create_relation(self, target) : nil
```

Comment:
Can this be expanded to a simple `if` statement please, for ease of following?

Reply:
done